### PR TITLE
Refactors CaseStudy out into varats-core

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 78
- Added 33
           

Complexity increasing per file
==============================
- varats/varats/paper_mgmt/paper_config.py  6
- varats/varats/utilss/cli_util.py  8
- varats-core/varats/base/sampling_method.py  3
- varats-core/varats/provider/cve/cve_provider.py  5
- varats-core/varats/provider/cve/cve_map.py  14
- varats/varats/paper_mgmt/case_study.py  7
- varats/varats/utilss/git_util.py  14
- varats-core/varats/report/report.py  4
- varats-core/varats/provider/bug/bug_provider.py  2
- varats-core/varats/provider/release/release_provider.py  10
- varats/varats/utilss/experiment_util.py  13
- varats/varats/utilss/bb_config.py  5
- varats/varats/paper_mgmt/paper_config_manager.py  11
- varats/varats/data/discover_reports.py  1
- varats-core/varats/paper/case_study.py  8
- varats/varats/paper_mgmt/artefacts.py  3
- varats-core/varats/provider/cve/cve.py  8
         

Clones added
============
- varats/varats/data/databases/blame_interaction_database.py  1
- varats/varats/utilss/git_util.py  2
- varats/varats/data/databases/blame_interaction_degree_database.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6180505&bid=20172618)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 78
- Added 33
           

Complexity increasing per file
==============================
- varats/varats/paper_mgmt/paper_config_manager.py  11
- varats/varats/paper_mgmt/artefacts.py  3
- varats-core/varats/provider/release/release_provider.py  10
- varats-core/varats/provider/bug/bug_provider.py  2
- varats-core/varats/base/sampling_method.py  3
- varats-core/varats/provider/cve/cve_map.py  14
- varats/varats/paper_mgmt/paper_config.py  6
- varats/varats/utilss/cli_util.py  8
- varats/varats/utilss/git_util.py  14
- varats-core/varats/paper/case_study.py  8
- varats-core/varats/report/report.py  4
- varats/varats/utilss/bb_config.py  5
- varats-core/varats/provider/cve/cve.py  8
- varats/varats/paper_mgmt/case_study.py  7
- varats-core/varats/provider/cve/cve_provider.py  5
- varats/varats/data/discover_reports.py  1
- varats/varats/utilss/experiment_util.py  13
         

Clones added
============
- varats/varats/data/databases/blame_interaction_database.py  1
- varats/varats/utilss/git_util.py  2
- varats/varats/data/databases/blame_interaction_degree_database.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6180505&bid=20172618)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 78
- Added 33
           

Complexity increasing per file
==============================
- varats-core/varats/provider/cve/cve_provider.py  5
- varats-core/varats/base/sampling_method.py  3
- varats-core/varats/provider/cve/cve.py  8
- varats/varats/utilss/cli_util.py  8
- varats-core/varats/provider/bug/bug_provider.py  2
- varats/varats/paper_mgmt/paper_config_manager.py  11
- varats/varats/paper_mgmt/case_study.py  7
- varats-core/varats/paper/case_study.py  8
- varats/varats/utilss/bb_config.py  5
- varats-core/varats/provider/release/release_provider.py  10
- varats/varats/utilss/git_util.py  14
- varats/varats/paper_mgmt/paper_config.py  6
- varats/varats/paper_mgmt/artefacts.py  3
- varats/varats/utilss/experiment_util.py  13
- varats/varats/data/discover_reports.py  1
- varats-core/varats/provider/cve/cve_map.py  14
- varats-core/varats/report/report.py  4
         

Clones added
============
- varats/varats/utilss/git_util.py  2
- varats/varats/data/databases/blame_interaction_database.py  1
- varats/varats/data/databases/blame_interaction_degree_database.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6180505&bid=20172618)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 79
- Added 33
           

Complexity increasing per file
==============================
- varats-core/varats/provider/cve/cve.py  8
- varats/varats/paper_mgmt/paper_config.py  6
- varats/varats/utilss/git_util.py  14
- varats/varats/paper_mgmt/artefacts.py  3
- varats-core/varats/provider/release/release_provider.py  10
- varats-core/varats/provider/cve/cve_provider.py  5
- varats-core/varats/provider/cve/cve_map.py  14
- varats-core/varats/paper/case_study.py  8
- varats/varats/paper_mgmt/case_study.py  7
- varats-core/varats/base/sampling_method.py  3
- varats/varats/data/discover_reports.py  1
- varats-core/varats/report/report.py  4
- varats/varats/utilss/experiment_util.py  13
- varats/varats/paper_mgmt/paper_config_manager.py  11
- varats/varats/utilss/cli_util.py  8
- varats/varats/utilss/bb_config.py  5
- varats-core/varats/provider/bug/bug_provider.py  2
         

Clones added
============
- varats/varats/utilss/git_util.py  2
- varats/varats/data/databases/blame_interaction_database.py  1
- varats/varats/data/databases/blame_interaction_degree_database.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6180505&bid=20172618)

File: docs/source/vara-ts-api/data.rst [hunk 1]
```diff
@@ -111,12 +111,12 @@ Data providers
 Providers are a means to supply additional data for a :ref:`project<Projects>`.
 For example, the :ref:`CVE provider` allows access to all CVEs that are related to a project.
 
-You can implement your own provider by creating a subclass of :class:`~varats.data.provider.provider.Provider` in its own subdirectory of ``data/provider``.
+You can implement your own provider by creating a subclass of :class:`~varats.provider.provider.Provider` in its own subdirectory of ``data/provider``.
```
Comment (boehmseb) [inline docs/source/vara-ts-api/data.rst]:
```suggestion
You can implement your own provider by creating a subclass of :class:`~varats.provider.provider.Provider` in its own subdirectory of ``provider`` in varats-core.
```

File: varats/varats/plots/blame_interaction_degree.py [hunk 2]
```diff
@@ -17,8 +17,9 @@
 from varats.plots.bug_annotation import draw_bugs
 from varats.plots.cve_annotation import draw_cves
 from varats.plots.plot import Plot, PlotDataEmpty
+#from varats.data.discover_reports import foo
```
Comment (boehmseb) [inline varats/varats/plots/blame_interaction_degree.py]:
```suggestion
```

File: varats/varats/plots/blame_lorenz_curve.py [hunk 3]
```diff
@@ -15,12 +15,13 @@
 from varats.data.reports.commit_report import CommitMap
 from varats.paper.case_study import CaseStudy
 from varats.plots.plot import Plot, PlotDataEmpty
+#from varats.data.discover_reports import foo
```
Comment (boehmseb) [inline varats/varats/plots/blame_lorenz_curve.py]:
```suggestion
```

File: varats/varats/tools/driver_casestudy.py [hunk 4]
```diff
@@ -10,28 +10,31 @@
 from argparse_utils import enum_action
 from plumbum import FG, colors, local
 
-from varats.data.provider.release.release_provider import ReleaseType
-from varats.data.report import FileStatusExtension, MetaReport
-from varats.paper import paper_config_manager as PCM
-from varats.paper.case_study import (
+from varats.base.sampling_method import SamplingMethod
+#from varats.data.discover_reports import foo
```
Comment (boehmseb) [inline varats/varats/tools/driver_casestudy.py]:
```suggestion
```

**Review by boehmseb — CHANGES_REQUESTED** [review]

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 79
- Added 33
           

Complexity increasing per file
==============================
- varats/varats/paper_mgmt/paper_config.py  6
- varats-core/varats/provider/bug/bug_provider.py  2
- varats/varats/data/discover_reports.py  1
- varats/varats/paper_mgmt/case_study.py  7
- varats/varats/paper_mgmt/artefacts.py  3
- varats/varats/paper_mgmt/paper_config_manager.py  11
- varats-core/varats/provider/cve/cve_map.py  14
- varats/varats/utilss/bb_config.py  5
- varats-core/varats/report/report.py  4
- varats/varats/utilss/git_util.py  14
- varats-core/varats/provider/release/release_provider.py  10
- varats/varats/utilss/cli_util.py  8
- varats/varats/utilss/experiment_util.py  13
- varats-core/varats/paper/case_study.py  8
- varats-core/varats/provider/cve/cve.py  8
- varats-core/varats/base/sampling_method.py  3
- varats-core/varats/provider/cve/cve_provider.py  5
         

Clones added
============
- varats/varats/data/databases/blame_interaction_database.py  1
- varats/varats/utilss/git_util.py  2
- varats/varats/data/databases/blame_interaction_degree_database.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6180505&bid=20172618)

**Review by boehmseb — APPROVED** [review]

**Pull request merged**